src/lib.rs

```diff
@@ -573,5 +573,41 @@ pub mod pallet {
 			Votes::<T>::remove(supersig_index, call_index);
 			UsersVotes::<T>::remove_prefix((supersig_index, call_index), None);
 		}
+
+		pub fn add_to_supersig(supersig_idx: u128, members: Vec<(T::AccountId, Roles)>) -> u128 {
+			let mut number_added = 0;
+			for (member, role) in members {
+				if Self::members(supersig_idx, &member) == Roles::NotMember {
+					Members::<T>::insert(supersig_idx, member, role);
+					number_added += 1;
+				}
+			}
+			MembersNumber::<T>::mutate(supersig_idx, |n| *n += number_added);
+
+			number_added
+		}
+
+		pub fn remove_from_supersig(
+			supersig_idx: u128,
+			members: Vec<T::AccountId>,
+		) -> Result<u128, pallet::Error<T>> {
+			let mut number_removed = 0;
+
+			let mut members = members.clone();
+			members.retain(|member| Self::members(supersig_idx, member) != Roles::NotMember);
+
+			if members.len() + 1 >= Self::members_number(supersig_idx) as usize {
+				return Err(Error::<T>::CannotRemoveUsers)
+			}
+			for member in members {
+				if Self::members(supersig_idx, &member) != Roles::NotMember {
+					Members::<T>::remove(supersig_idx, member);
+					number_removed += 1;
+				}
+			}
```

Comment:
You can do this in one single loop:
loop over `members`
use try_mutate
If the value is `NotMember`, don't mutate and go next iteration
if value is not `NotMember`:
- increase `number_removed`
- check `total_members - number_removed >= 1`, if not revert transaction, else remove key from storage

We loop only once.

Reply:
In a closure (so in a try_mutate), if i return an error, it will not leave the function... so if i have 5 users in the supersig, and want to remove 4, the first three will be removed, and then it will block for the fourth

What i intended to do is that if you want to remove 4 of 5 users, the function will throw an error without removing any users.

Or i can make the function transactional, but if i do, i dont even need to mkae a try_mutate, i can just do the check in my `if ... != Roles::NotMember`